engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -94,16 +94,35 @@ struct avengers_shield_base_t : public paladin_spell_t
 struct avengers_shield_dt_t : public avengers_shield_base_t
 {
   avengers_shield_dt_t( paladin_t* p ) :
-    avengers_shield_base_t( "avengers_shield_divine_toll", p, p -> find_specialization_spell( "Avenger's Shield" ), "" )
+    avengers_shield_base_t( "avengers_shield_divine_toll", p, "" )
   {
     background = true;
   }
 };
 
+struct avengers_shield_dr_t : public avengers_shield_base_t
+{
+  avengers_shield_dr_t( paladin_t* p ):
+    avengers_shield_base_t( "avengers_shield_divine_resonance", p, "" )
+  {
+    background = true;
+  }
+
+  double action_multiplier() const override
+  {
+    // 2021-06-26 Divine resonance damage is increased by Moment of Glory, but
+    // does not consume the buff.
```

Comment:
If it's not consumed you'll have to adjust or override avenger's shield's execute code to match that. I suggest checking that the action is background=false before decrementing moment of glory

Reply:
Divine resonance and divine toll (and ringing clarity for that matter) both inherit from avengers_shield_base_t which does not have the moment of glory consumption
Avenger's shield the active ability is its own subclass of that base with the moment of glory implementation (and other "active ability" stuff)

Reply:
As a python guy I prefer that sub-classing to the way Shield of the Righteous does it with overloaded functions, but idk which is best practice.

Reply:
Makes sense, I completely overlooked that